react/features/share-room/actions.js

```diff
@@ -0,0 +1,66 @@
+/* @flow */
+
+import { BEGIN_SHARE_ROOM, END_SHARE_ROOM } from './actionTypes';
+
+
+/**
+ * Begin to share the room URL for the current conference.
+ *
+ * @param  {string} roomUrl - URL of the room to be shared.
+ * @returns {Function}
+ */
+export function beginShareRoom(roomUrl: ?string): Function {
+    return (dispatch, getState) => {
+        if (!roomUrl) {
+            const { conference, room } = getState()['features/base/conference'];
+
+            // eslint-disable-next-line no-param-reassign
+            roomUrl = _getRoomUrl(conference, room);
+        }
+
+        if (roomUrl) {
+            dispatch({
+                type: BEGIN_SHARE_ROOM,
+                roomUrl
+            });
+        }
+    };
+}
+
+
+/**
+ * Signals that the room URL sharing process has finished.
+ *
+ * @param {string} roomUrl - URL of the room which was shared.
+ * @param {boolean} shared - Set to true if the URL was shared successfully,
+ * false otherwise.
+ * @returns {{
+ *      type: END_SHARE_ROOM,
+ *      roomUrl: string,
+ *      shared: boolean
+ * }}
+ */
+export function endShareRoom(roomUrl: string, shared: boolean): Object {
+    return {
+        type: END_SHARE_ROOM,
+        roomUrl,
+        shared
+    };
+}
+
+
+/**
+ * Gets the public URL for a conference.
+ *
+ * @param  {JitsiConference} conference - Conference object.
+ * @param  {string} room - Room identifier.
+ * @private
+ * @returns {string|null}
+ */
+function _getRoomUrl(conference, room) {
+    if (!conference || !room) {
+        return null;
+    }
+
+    return `https://${conference.connection.options.hosts.domain}/${room}`;
```

Comment:
@lyubomir I just realized this is not going to work with deployments which serve meet in a sub-directory, do you remember where we store that piece of information?